controllers/org_billingentity_name_cache_controller.go

```diff
@@ -0,0 +1,77 @@
+package controllers
+
+import (
+	"context"
+	"math/rand"
+	"time"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	billingv1 "github.com/appuio/control-api/apis/billing/v1"
+	orgv1 "github.com/appuio/control-api/apis/organization/v1"
+)
+
+// OrgBillingEntityNameCacheController reconciles OrganizationMembers resources
+type OrgBillingEntityNameCacheController struct {
+	client.Client
+	Recorder record.EventRecorder
+	Scheme   *runtime.Scheme
+
+	RefreshInterval time.Duration
+	RefreshJitter   time.Duration
+}
+
+//+kubebuilder:rbac:groups="organization.appuio.io",resources=organizations,verbs=get;list;watch
+//+kubebuilder:rbac:groups="organization.appuio.io",resources=organizations/status,verbs=get;update;patch
+
+// Reconcile reacts on changes of users and mirrors these changes to Keycloak
+func (r *OrgBillingEntityNameCacheController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := log.FromContext(ctx)
+	log.V(4).WithValues("request", req).Info("Reconciling")
+
+	var org orgv1.Organization
+	if err := r.Get(ctx, req.NamespacedName, &org); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if !org.ObjectMeta.DeletionTimestamp.IsZero() {
+		return ctrl.Result{}, nil
+	}
+
+	if org.Spec.BillingEntityRef == "" {
+		return ctrl.Result{}, nil
+	}
+
+	var be billingv1.BillingEntity
+	err := r.Client.Get(ctx, client.ObjectKey{Name: org.Spec.BillingEntityRef}, &be)
+	if err != nil {
+		return ctrl.Result{}, err
```

Comment:
Nitpick: I'm not sure if we want to log if the error is `not found`.

Reply:
Not found is an important error IMO since we validate that the entity exists on linking them.

Reply:
Hmm, maybe you're right. I'm not sure how helpful it really is if we log `not found` every 5 min if someone deletes the BE in Odoo, but you're right it's a relevant error.